Replace Hook with MethodHook where possible

deathbeam opened the pull request:
In order to cleanup the Hooks class a bit, replace runescape-client
Hook with MethodHook in mixins where possible/convenient.

Depends on !13
Depends on !14

Signed-off-by: Tomas Slusny <slusnucky@gmail.com>

Adam- commented:
This should probably have one commit per method moved. I suspect there will be things which no longer have to be exposed in rl-api as a result of this. Also the entire `@Hook` annotation can go away.

deathbeam commented:
Alright. So that means for the performance-critical methods that cannot be just sent through eventBus i should just call methods in Hooks directly from mixins? In similar way as mouse/key listeners do atm?

deathbeam commented:
Split each hook change to single commit, updated !13 with removal of `@Hook` annotation and used constructor injection for remaining hooks, so I call this update feature complete :d

Adam- commented on runelite-client/src/main/java/net/runelite/client/callback/Hooks.java:
```diff
@@ -467,31 +464,6 @@ public static void menuOpened(Client client, int var1, int var2)
 		eventBus.post(event);
 	}
 
-	/**
```
Move the javadoc too.

Adam- commented on runelite-mixins/src/main/java/net/runelite/mixins/RSClientMixin.java:
```diff
@@ -1054,4 +1055,18 @@ public static void updateNpcs(boolean var0, PacketBuffer var1)
 	}
+
+	@Inject
+	@FieldHook("focused")
```
Any reason why this isnt a methodhook like before?

deathbeam replied:
No reason for it to be methodHook when that field is properly modified in both focusLost and focusGained, this felt like simpler approach.

Adam- commented on runelite-mixins/src/main/java/net/runelite/mixins/RSItemCompositionMixin.java:
```diff
@@ -73,4 +76,13 @@ public void resetShiftClickActionIndex()
+	@MethodHook(value = "post", end = true)
+	public void post()
+	{
+		final  PostItemComposition event = new PostItemComposition();
```
Fix formatting

deathbeam replied:
For the annotation?

Adam- commented on runelite-client/src/main/java/net/runelite/client/callback/Hooks.java:
```diff
@@ -383,27 +380,6 @@ public static void addChatMessage(int type, String name, String message, String
 		eventBus.post(chatMessage);
 	}
 
-	/**
```
Do not remove javadoc.

Adam- commented on runelite-client/src/main/java/net/runelite/client/callback/Hooks.java:
```diff
@@ -367,19 +365,6 @@ public static boolean menuActionHook(int actionParam, int widgetId, int menuActi
 
-	public static void addChatMessage(int type, String name, String message, String sender)
-	{
-		if (log.isDebugEnabled())
```
Any reason this is removed? The Hooks logger is accessible. We could also maybe just inject a logger.

Adam- commented on runelite-mixins/src/main/java/net/runelite/mixins/RSMainBufferProviderMixin.java:
```diff
@@ -42,6 +42,6 @@
 	final void draw(Graphics graphics, int x, int y)
 	{
-		Hooks.draw(this, graphics, x, y);
+		Hooks.draw(this, graphics);
```
What is the point of this commit?

deathbeam replied:
Consistency with other draw-related methods, it does not need x/y so only needed fields should be exposed
